tests/libnode/node.go

```diff
@@ -122,56 +123,94 @@ func GetNodeDrainKey() string {
 	return virtconfig.NodeDrainTaintDefaultKey
 }
 
-func addLabelAnnotationHelper(nodeName, key, value string, isLabel bool) {
+type mapType string
+
+const (
+	label      mapType = "label"
+	annotation mapType = "annotation"
+)
+
+type mapAction string
+
+const (
+	add    mapAction = "add"
+	remove mapAction = "remove"
+)
+
+// Adds or removes a label or annotation from a node. When removing a label/annotation, the "value" parameter
+// is ignored.
+func addRemoveLabelAnnotationHelper(nodeName, key, value string, mapType mapType, mapAction mapAction) {
```

Comment:
```suggestion
it is a stepo into the right direction, WDYT about this?

func patchNodeMapHelper(nodeName string, type mapType, 
fetchMap func(node *k8sv1.Node) map[string]string,
mutateMap func(key, val string, m map[string]string) map[string]string) {

```
and use it like this:
```
fetchLabelsFunc = func(node *k8sv1.Node) map[string]string { return node.Labels }
fetchAnnotationsFunc = func(node *k8sv1.Node) map[string]string { return node.Annotations }

addKvFunc = func(key, value string)  	func(m map[string]string) map[string]string  {
return func(m map[string]string) map[string]string {
			m[key] = val
			return m
		}
}

func AddLabelToNode(nodeName, key, value string) {
	addRemoveLabelAnnotationHelper(nodeName, label,
	fetchLabels, 
	addKvFunc(key, value))
}

func AddAnnotationToNode(nodeName, key, value string) {
	addRemoveLabelAnnotationHelper(nodeName, annotation
	fetchAnnotationsFunc, 
	addKvFunc(key, value))
}
```

Reply:
It is a pity that `go` does not have nice functional syntax sugar for anonymous functions

Reply:
I thought about doing it in the way you described, but I'm not sure it is better. My thinking was that these function are exclusively related to `addRemoveLabelAnnotationHelper()` function, so why expose them into the global outer scope?

I think this way is better since it's obvious what's the purpose of these functions and the scope which they serve.

Reply:
> It is a pity that go does not have nice functional syntax sugar for anonymous functions

Indeed!!